Makefile

```diff
@@ -1,132 +1,107 @@
+
+
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+kustomize: ## Download kustomize locally if necessary.
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
 
-.PHONY: lint/golangci-lint
-lint/golangci-lint: tools/golangci-lint .golangci.yml  ## Run golangci-lint.
-	golangci-lint run
-
-# Build the docker image
-docker-build:
-	docker image build . -t ${IMG}
-
-# Push the docker image
-docker-push:
-	docker image push ${IMG}
-
-kubebuilder: ${KUBEBUILDER_ASSETS}/kubebuilder
-${KUBEBUILDER_ASSETS}/kubebuilder:
-	@{ \
-		set -e ;\
-		TMP_DIR=./tmp/kubebuilder ;\
-		mkdir -p $$TMP_DIR ;\
-		curl -sSL https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${KUBEBUILDER_VERSION}/kubebuilder_${KUBEBUILDER_VERSION}_${GOOS}_${GOARCH}.tar.gz | tar -xz -C $$TMP_DIR ;\
-		mv $$TMP_DIR/kubebuilder_${KUBEBUILDER_VERSION}_${GOOS}_${GOARCH} ${KUBEBUILDER_ASSETS_DIR} ;\
-		rm -rf $$TMP_DIR ;\
-	}
-
-.PHONY: run
-run: $(SKAFFOLD)
-	# NOTE: since skaffold uses kind executable from PATH directly, so this overrides PATH to use project local executable.
-	@PATH=$${PWD}/bin:$${PATH} $(SKAFFOLD) dev --filename=./skaffold/skaffold.yaml
-
-.PHONY: manifests
-manifests: $(CONTROLLER_GEN)
-	@$(CONTROLLER_GEN) crd:trivialVersions=true rbac:roleName=manager-role webhook paths="./pkg/api/..." output:crd:artifacts:config=config/crd/bases
-
-.PHONY: deepcopy
-deepcopy: $(CONTROLLER_GEN)
-	 @$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
-
-controlller-gen: $(CONTROLLER_GEN)
-$(CONTROLLER_GEN): go.sum
-	@go build -o $(CONTROLLER_GEN) sigs.k8s.io/controller-tools/cmd/controller-gen
-
-kind: $(KIND)
-$(KIND): go.sum
-	@go build -o ./bin/kind sigs.k8s.io/kind
-
-kubectl: $(KUBECTL)
-$(KUBECTL): .kubectl-version
-	@curl -Lso $(KUBECTL) https://storage.googleapis.com/kubernetes-release/release/$(shell cat .kubectl-version)/bin/$(GOOS)/$(GOARCH)/kubectl
-	@chmod +x $(KUBECTL)
-
-skaffold: $(SKAFFOLD)
-$(SKAFFOLD): .skaffold-version
-	@curl -Lso $(SKAFFOLD) https://storage.googleapis.com/skaffold/releases/$(shell cat .skaffold-version)/skaffold-$(GOOS)-$(GOARCH)
-	@chmod +x $(SKAFFOLD)
-
-kpt: $(KPT)
-$(KPT): go.sum
-	@go build -o $(KPT) github.com/GoogleContainerTools/kpt
-
-.PHONY: kind-cluster
-kind-cluster: $(KIND) $(KUBECTL)
-	@$(KIND) delete cluster --name spanner-autoscaler
-	@$(KIND) create cluster --name spanner-autoscaler --image kindest/node:v${KIND_NODE_VERSION}
-	@$(KUBECTL) apply -f ./config/crd/bases/spanner.mercari.com_spannerautoscalers.yaml
-	@$(KUBECTL) apply -f ./kind/namespace.yaml
-	@$(KUBECTL) apply -f ./kind/rbac
-
-.PHONY: kpt/crd
-kpt/crd: manifests kpt
-	@kustomize build config/crd > ./kpt/crd/spannerautoscalers.spanner.mercari.com.yaml
```

Comment:
[Ref](https://github.com/mercari/spanner-autoscaler/pull/40#issuecomment-916685279)
This is the current status of the `Makefile`:
<img width="776" alt="Screen Shot 2021-10-13 at 18 25 01" src="https://user-images.githubusercontent.com/1299606/137106105-7cfa8f67-494a-400c-8ba9-4f9cb3586352.png">

- The current `kpt` package was anyways broken (#41), so I didn't bother on porting it over yet. I will fix it along with `kpt` upgrade in #43.

- As for others:
  - `kubectl`: I think we don't need to install it on the developer's system because this is a basic tool for interaction with k8s and is expected to be present already.
  - `kind`: Same as above. I think any k8s developer would already have access to some kind of development k8s environment (be it `kind` or any other k8s setup of their choice).
  - `kubebuilder`: Same as above. Anyone building or modifying operators, would already have it on their system I think. We shouldn't burden ourselves with maintaining more code/components than what is/are necessary.
  - `skaffold`: I'm not sure about this one. Is this really used by any of the current code owners for development? If so, then I can port it.

Let me know what you think and which (or all) targets should be ported over.